patches/api/0387-Paper-Plugins.patch

```diff
@@ -0,0 +1,963 @@
+index b8623575b1c1b565560c2dd6438190716845a652..785952d60df7c0ab7b247ba81d1970b5a90e5340 100644
+--- a/src/main/java/org/bukkit/command/SimpleCommandMap.java
++++ b/src/main/java/org/bukkit/command/SimpleCommandMap.java
+@@ -33,7 +33,7 @@ public class SimpleCommandMap implements CommandMap {
+     private void setDefaultCommands() {
+         register("bukkit", new VersionCommand("version"));
+         register("bukkit", new ReloadCommand("reload"));
+-        register("bukkit", new PluginsCommand("plugins"));
++        //register("bukkit", new PluginsCommand("plugins")); Paper
+         register("bukkit", new co.aikar.timings.TimingsCommand("timings")); // Paper
+     }
+ 
+diff --git a/src/main/java/org/bukkit/plugin/Plugin.java b/src/main/java/org/bukkit/plugin/Plugin.java
+index 08aef59d8443038771704d9587e31f299e587307..3948cebb552ae1873421a8354c686a0b4afff3a7 100644
+--- a/src/main/java/org/bukkit/plugin/Plugin.java
++++ b/src/main/java/org/bukkit/plugin/Plugin.java
+@@ -34,6 +34,8 @@ public interface Plugin extends TabExecutor {
+     @NotNull
+     public PluginDescriptionFile getDescription();
+ 
++    io.papermc.paper.plugin.provider.configuration.PluginConfiguration getConfiguration();
++
+     /**
+      * Gets a {@link FileConfiguration} for this plugin, read through
+      * "config.yml"
+diff --git a/src/main/java/org/bukkit/plugin/PluginDescriptionFile.java b/src/main/java/org/bukkit/plugin/PluginDescriptionFile.java
+index 0c9f4d1e9104fa6951114c1f9ec954dfcc749196..f6193c406ee7ba6b7ed66d1aca20da35150bddad 100644
+--- a/src/main/java/org/bukkit/plugin/PluginDescriptionFile.java
++++ b/src/main/java/org/bukkit/plugin/PluginDescriptionFile.java
+@@ -194,7 +194,7 @@ import org.yaml.snakeyaml.nodes.Tag;
+  *      inferno.burningdeaths: true
+  *</pre></blockquote>
+  */
+-public final class PluginDescriptionFile {
++public class PluginDescriptionFile implements io.papermc.paper.plugin.provider.configuration.BukkitConfigurationWrapper { // Paper
+     private static final Pattern VALID_NAME = Pattern.compile("^[A-Za-z0-9 _.-]+$");
+     private static final ThreadLocal<Yaml> YAML = new ThreadLocal<Yaml>() {
+         @Override
+@@ -254,6 +254,31 @@ public final class PluginDescriptionFile {
+     private Set<PluginAwareness> awareness = ImmutableSet.of();
+     private String apiVersion = null;
+     private List<String> libraries = ImmutableList.of();
++    // Paper start - oh my goddddd
++    public PluginDescriptionFile(String rawName, String name, List<String> provides, String main, String classLoaderOf, List<String> depend, List<String> softDepend, List<String> loadBefore, String version, Map<String, Map<String, Object>> commands, String description, List<String> authors, List<String> contributors, String website, String prefix, PluginLoadOrder order, Map<?, ?> lazyPermissions, PermissionDefault defaultPerm, Set<PluginAwareness> awareness, String apiVersion, List<String> libraries) {
```

Comment:
Don't put all params on the same line, it's not an autobus, move at least these stuff in three rows

Reply:
It's auto generated and internal, I don't want people to use that and I am not sure if I am gonna keep that.